bundle/src/main/java/com/adobe/acs/commons/redirects/filter/RedirectFilter.java

```diff
@@ -0,0 +1,508 @@
+        Map<Pattern, RedirectRule> patternRules = new LinkedHashMap<>();
+        long t0 = System.currentTimeMillis();
+        try (ResourceResolver resolver = resourceResolverFactory.getServiceResourceResolver(
+                Collections.singletonMap(ResourceResolverFactory.SUBSERVICE, SERVICE_NAME))) {
+            Resource storageResource = resolver.getResource(getStoragePath());
+            if (storageResource != null) {
+                Collection<RedirectRule> rules = getRules(storageResource);
+                for (RedirectRule rule : rules) {
+                    if (rule.getRegex() != null) {
+                        patternRules.put(rule.getRegex(), rule);
+                    } else {
+                        pathRules.put(rule.getSource(), rule);
+                    }
+                }
+            }
+        } catch (LoginException e) {
+            log.error("Failed to get resolver for {}", SERVICE_NAME, e);
+        }
+        synchronized (this) {
+            this.pathRules = pathRules;
+            this.patternRules = patternRules;
+        }
+        log.debug("{} rules loaded in {} ms", pathRules.size() + patternRules.size(),
+                System.currentTimeMillis() - t0);
+    }
+
+    Map<String, RedirectRule> getPathRules() {
+        return pathRules;
+    }
+
+    Map<Pattern, RedirectRule> getPatternRules() {
+        return patternRules;
+    }
+
+    /**
+     * Read redirect configurations from the repository, i.e.
+     *  /var/acs-commons/redirects --> Collection<RedirectRule>
+     *
+     * @param resource the parent resource containing redirect configurations
+     * @return a list of redirect configurations . Can be empty if no redirects are
+     * configured.
+     */
+    public static Collection<RedirectRule> getRules(Resource resource) {
+        Collection<RedirectRule> rules = new ArrayList<>();
+        for (Resource res : resource.getChildren()) {
+            if(REDIRECT_RULE_RESOURCE_TYPE.equals(res.getResourceType())){
+                try {
+                    rules.add(new RedirectRule(res.getValueMap()));
+                } catch (Exception e) {
+                    log.error("Invalid redirect configuration:  {}", resource.getPath(), e);
+                }
+            }
+        }
+        return rules;
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        if (!(request instanceof SlingHttpServletRequest)
+                || !(response instanceof SlingHttpServletResponse)) {
+            chain.doFilter(request, response);
+            return;
+        }
+
+        SlingHttpServletRequest slingRequest = (SlingHttpServletRequest) request;
+        SlingHttpServletResponse slingResponse = (SlingHttpServletResponse) response;
+
+        if (isEnabled() && doesRequestMatch(slingRequest) && handleRedirect(slingRequest, slingResponse)) {
+            return;

```

Comment:
Some debug log to improve debugging possibilities would help here.

Reply:
The redirection fact is logged in handleRedirect:
`
               log.info("Redirecting {} to {}, statusCode: {}",
                        resourcePath, location, redirectRule.getStatusCode());
`
people can aggregate it in Splunk and build reports of popular redirects and redirects per day, etc. There are other log.info and log.debug statements across the code which help with debugging.